packages/core/src/codewhisperer/util/supplementalContext/crossFileContextUtil.ts

```diff
@@ -132,14 +132,20 @@ export async function fetchSupplementalContextForSrcV1(
 
     // Step 4: Transform best chunks to supplemental contexts
     const supplementalContexts: CodeWhispererSupplementalContextItem[] = []
+    let totalLength = 0
     for (const chunk of bestChunks) {
         throwIfCancelled(cancellationToken)
 
+        if (totalLength > crossFileContextConfig.maximumTotalLength) {
+            break
+        }
+
```

Comment:
Should this go after the counter line? Otherwise we could have already went past the max.

Reply:
yea that's a good callout, Max

Reply:
updated